src/cbc_sdk/platform/observations.py

```diff
@@ -0,0 +1,548 @@
+
+        while still_fetching:
+            result_url = "{}?start={}&rows={}".format(
+                result_url_template, current, self._batch_size
+            )
+            result = self._cb.get_object(result_url, query_parameters=query_parameters)
+            results = result.get("results", [])
+
+            self._total_results = result.get("num_available", 0)
+            self._count_valid = True
+
+            for item in results:
+                yield item
+                current += 1
+                rows_fetched += 1
+
+                if rows and rows_fetched >= rows:
+                    still_fetching = False
+                    break
+
+            if current >= self._total_results:
+                still_fetching = False
+
+            log.debug("current: {}, total_results: {}".format(current, self._total_results))
+
+    def get_group_results(
+        self,
+        fields,
+        max_events_per_group=None,
+        rows=500,
+        start=None,
+        range_duration=None,
+        range_field=None,
+        range_method=None
+    ):
+        """
+        Get group results grouped by provided fields.
+
+        Args:
+            fields (str / list): field or fields by which to perform the grouping
+            max_events_per_group (int):Maximum number of events in a group, if not provided, all events will be returned
+            rows (int): Number of rows to request, can be paginated
+            start (int): First row to use for pagination
+            ranges (dict): dict with information about duration, field, method
+
+        Returns:
+            dict: grouped results
+        """
+        if not isinstance(fields, list) and not isinstance(fields, str):
+            raise ApiError("Fields should be either a single field or list of fields")
+
+        if isinstance(fields, str):
+            fields = [fields]
+
+        if not all((gf in ObservationQuery.VALID_GROUP_FIELDS) for gf in fields):
+            raise ApiError("One or more invalid aggregation fields")
+
+        if not self._query_token:
+            self._submit()
+
+        result_url = "/api/investigate/v2/orgs/{}/observations/search_jobs/{}/group_results".format(
+            self._cb.credentials.org_key,
+            self._query_token,
+        )
+
+        data = self._build_grouped_body(
+            fields,
+            max_events_per_group,
+            rows,
+            start,
+            range_duration,
+            range_field,
+            range_method
+        )
+
+        still_fetching = True
+        while still_fetching:
+            result = self._cb.post_object(result_url, data).json()
+            contacted = result.get("contacted", 0)
+            completed = result.get("completed", 0)
+            if contacted < completed:
+                time.sleep(0.5)
+                continue
+            else:
+                still_fetching = False
+
+        for group in result.get("group_results", []):
+            yield group
```

Comment:
Let's avoid returning the raw json here as we provide helper functions on the Observation Class. I would recommend creating an ObservationGroup class which contains a list of Observations. If you follow up where the `_search` class gets invoked that calls `yield self._doc_class.new_object(self._cb, item)` for context

Reply:
Done.